crates/matrix-sdk-ui/src/notification_client.rs

```diff
@@ -148,11 +169,50 @@ impl NotificationClient {
         //
         // Keep timeouts small for both, since we might be short on time.
 
-        let with_locking = WithLocking::from(self.with_cross_process_lock);
+        let with_locking = WithLocking::from(matches!(
+            self.process_setup,
+            NotificationProcessSetup::MultipleProcesses
+        ));
+
+        let sync_permit_guard = match &self.process_setup {
+            NotificationProcessSetup::MultipleProcesses => {
+                // We're running on our own process, dedicated for notifications. In that case,
+                // create a dummy sync permit. That works under the assumption that there aren't
+                // multiple calls to this function at the same time (i.e. it's not reentrant).
+                let sync_permit = Arc::new(AsyncMutex::new(EncryptionSyncPermit::new()));
+                sync_permit.lock_owned().await
+            }
+
+            NotificationProcessSetup::SingleProcess { sync_service } => {
+                if let Some(permit_guard) = sync_service.try_get_encryption_sync_permit() {
+                    permit_guard
+                } else {
+                    // There's already a sync service active, thus the encryption sync is already
+                    // running elsewhere. As a matter of fact, if the event was encrypted, that
+                    // means we were racing against the encryption sync. Wait a bit, attempt to
+                    // decrypt, and carry on.
+
+                    let mut wait = 200; // we get to wait 7 times that number at most.
```

Comment:
The comment looks incorrect. You're retrying 3 times here, no?

Reply:
I'm retrying 3 times, but the comment says "7 times *that number*", since we're doubling the amount of time waited, we're waiting for it 1 + 2 + 4 = 7 times. I'll expand the comment.